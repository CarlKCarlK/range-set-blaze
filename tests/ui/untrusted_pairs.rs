use range_set_blaze::RangeSetBlaze;

fn main() {
    let guaranteed = RangeSetBlaze::from_iter([1..=2, 3..=4, 5..=6]).into_ranges();
    let _range_set_blaze = RangeSetBlaze::from_sorted_disjoint(guaranteed); // yep
    let not_guaranteed = [5..=6, 1..=3, 3..=4].into_iter();
    let _range_set_blaze = RangeSetBlaze::from_sorted_disjoint(not_guaranteed); // nope
}
