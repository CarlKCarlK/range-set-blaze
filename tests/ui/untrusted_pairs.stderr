error[E0277]: the trait bound `std::array::IntoIter<RangeInclusive<{integer}>, 3>: SortedDisjoint<_>` is not satisfied
 --> tests/ui/untrusted_pairs.rs:7:62
  |
7 |     let _range_set_int = RangeSetBlaze::from_sorted_disjoint(not_guaranteed); // nope
  |                          ----------------------------------- ^^^^^^^^^^^^^^ the trait `SortedDisjoint<_>` is not implemented for `std::array::IntoIter<RangeInclusive<{integer}>, 3>`
  |                          |
  |                          required by a bound introduced by this call
  |
  = help: the following other types implement trait `SortedDisjoint<T>`:
            CheckSortedDisjoint<T, I>
            DynSortedDisjoint<'a, T>
            IntoRangesIter<T>
            NotIter<T, I>
            RangesIter<'_, T>
            UnionIter<T, I>
            itertools::tee::Tee<I>
note: required by a bound in `RangeSetBlaze::<T>::from_sorted_disjoint`
 --> src/lib.rs
  |
  |         I: SortedDisjoint<T>,
  |            ^^^^^^^^^^^^^^^^^ required by this bound in `RangeSetBlaze::<T>::from_sorted_disjoint`
